### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by azangru**
## Description
Routine dependencies update prompted by the fact that React core team has announced the release of React 19 beta. They also released React 18.3, which is a stepping stone to React 19 (see the [upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide)) 

Notable changes in this PR:
- Updated React v18.2 -> 18.3.1
- Switched to the new JSX transform (see [docs](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html)), which was introduced back in 2020, but to which we never migrated. According to the release notice about React 19 beta, the new transform is going to be a requirement for React 19
  - Removed React imports from all react files, because, with the new transform, they are no longer needed

## Deployment URL(s)
http://update-dependencies.review.ensembl.org

### CODE REVIEW

File: src/shared/components/pointer-box/PointerBox.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { ReactNode, useRef, useEffect, useState } from 'react';
+import { ReactNode, useRef, useEffect, useState } from 'react';
```
Comment (imransl) [inline src/shared/components/pointer-box/PointerBox.tsx:17]: Shouldn't  `ReactNode` be a type import?

File: src/shared/components/privacy-banner/PrivacyBanner.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { FunctionComponent } from 'react';
+import { FunctionComponent } from 'react';
```
Comment (imransl) [inline src/shared/components/privacy-banner/PrivacyBanner.tsx:17]: Type import?

File: src/shared/components/pointer-box/PointerBox.tsx (hunk 1 above)
Comment (azangru) [reply src/shared/components/pointer-box/PointerBox.tsx:17]: I am sure this was the automated script 😄

File: src/shared/components/round-button/RoundButton.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { ReactNode } from 'react';
+import { ReactNode } from 'react';
```
Comment (imransl) [inline src/shared/components/round-button/RoundButton.tsx:17]: Another type import?

File: src/shared/components/simple-select/SimpleSelect.test.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { useRef, useEffect, useState, ChangeEvent } from 'react';
+import { useRef, useEffect, useState, ChangeEvent } from 'react';
```
Comment (imransl) [inline src/shared/components/simple-select/SimpleSelect.test.tsx:17]: Type import (`ChangeEvent`)

File: src/shared/components/table/ColumnHead.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { DetailedHTMLProps, ThHTMLAttributes } from 'react';
+import { DetailedHTMLProps, ThHTMLAttributes } from 'react';
```
Comment (imransl) [inline src/shared/components/table/ColumnHead.tsx:17]: I presume these should be type imports too?

File: src/shared/components/text-line/TextLine.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { ReactNode, HTMLAttributes } from 'react';
+import { ReactNode, HTMLAttributes } from 'react';
```
Comment (imransl) [inline src/shared/components/text-line/TextLine.tsx:17]: Type imports?

File: src/shared/components/textarea/ShadedTextarea.test.tsx [hunk 7]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { useRef, MutableRefObject } from 'react';
+import { useRef, MutableRefObject } from 'react';
```
Comment (imransl) [inline src/shared/components/textarea/ShadedTextarea.test.tsx:17]: Type import

File: src/shared/components/textarea/ShadedTextarea.tsx [hunk 8]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { forwardRef, ForwardedRef } from 'react';
+import { forwardRef, ForwardedRef } from 'react';
```
Comment (imransl) [inline src/shared/components/textarea/ShadedTextarea.tsx:17]: type import

File: src/shared/components/view-in-app-popup/ViewInAppPopup.tsx [hunk 9]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { useState, useRef, ReactNode } from 'react';
+import { useState, useRef, ReactNode } from 'react';
```
Comment (imransl) [inline src/shared/components/view-in-app-popup/ViewInAppPopup.tsx:17]: One more here

File: src/shared/components/tooltip/Tooltip.tsx [hunk 10]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { ReactNode, useEffect, useState } from 'react';
+import { ReactNode, useEffect, useState } from 'react';
```
Comment (imransl) [inline src/shared/components/tooltip/Tooltip.tsx:17]: Another here

File: src/shared/components/textarea/Textarea.tsx [hunk 11]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import React, { forwardRef, TextareaHTMLAttributes, ForwardedRef } from 'react';
+import { forwardRef, TextareaHTMLAttributes, ForwardedRef } from 'react';
```
Comment (imransl) [inline src/shared/components/textarea/Textarea.tsx:17]: Type imports here and below too

**Comment by imransl** [comment]
Happy about most of the PR. There are a few type imports that have been missed out and the comments have only been about them.

**Review by imransl — APPROVED** [review]

**Pull request merged**